Tools/src/Tools/HgcrocEmulator.cxx

```diff
@@ -221,8 +221,8 @@ bool HgcrocEmulator::digitize(
     }  // TOT or ADC Mode
   }  // sampling baskets
 
-
-  if(savePulseTruthInfo_) pulseTruthColl_->push_back(ldmx::HgcrocPulseTruth(channelID, pulse));
+  if (savePulseTruthInfo_)
+    pulseTruthColl_->push_back(ldmx::HgcrocPulseTruth(channelID, pulse));
```

Comment:
So, I _think_ (haven't checked) that this vector of HgcrocPulseTruth never gets cleared. Am I wrong? If not, we will want to clear it at the start of each event so that there aren't pulses from previous events in the collection for a given event.

Reply:
The collection is declared inside the producer, so it's recreated/cleared for each event.